packages/gasket-cli/src/scaffold/actions/generate-files.js

```diff
@@ -1,78 +1,128 @@
+const debug = require('diagnostics')('gasket:cli:generate-files');
+const fs = require('fs').promises;
 const { promisify } = require('util');
 const Handlebars = require('handlebars');
-// TODO: See if we can move away from pump to a async solution
-const pump = promisify(require('pump'));
-const map = require('map-stream');
 const path = require('path');
+const mkdirp = require('mkdirp');
+const { dim } = require('chalk');
 const action = require('../action-wrapper');
 
+const glob = promisify(require('glob'));
+
 /**
- * Generate the app files and templates using context
+ * Find all duplicate target files and reduce to single descriptor.
+ * Keeps track of overrides.
+ * Last in wins.
+ *
+ * @param {object[]} descriptors - Details about files to generate
+ * @returns {object[]} descriptors
+ */
+function reduceDescriptors(descriptors) {
+  const reduced = descriptors.reduce((acc, desc) => {
+    const { target } = desc;
+    if (acc.has(target)) {
+      const prev = acc.get(target);
+      if (prev.from !== desc.from) {
+        debug(`Using "${ target }" from ${ desc.from } instead of ${ prev.from }`);
+        desc.overrides = prev.from;
+      }
+    }
+    acc.set(target, desc);
+    return acc;
+  }, new Map());
+
+  return Array.from(reduced.values());
+}
+
+/**
+ * Build a list of descriptions of all files we want to generate
  *
  * @param {CreateContext} context - Create context
- * @param {Spinner} spinner - Spinner
- * @returns {Promise} promise
+ * @returns {object[]} descriptors
  */
-async function generateFiles(context, spinner) {
-  const { dest, files, generatedFiles } = context;
+async function getDescriptors(context) {
+  const { dest, files } = context;
+  const descriptors = (
+    await Promise.all(files.globSets.map(async set => {
+      const { globs, source } = set;
+      const matches = await Promise.all(globs.map(async pattern => {
+        const srcPaths = await glob(pattern, { nodir: true });
+        const base = path.resolve(pattern.replace(/\/(.)?\*(.+)?/, ''));
```

Comment:
What is this pattern supposed to match? Is that `(.)` correct or is it supposed to be:

```suggestion
        const base = path.resolve(pattern.replace(/\/\.?\*.*/, ''));
```